### ISSUE TITLE
Automatically transform dates to a consistent format

### ISSUE THREAD

**Issue by apprehensions** [you-apps/ConnectYou#430]
### Describe your suggested feature

In my contacts i have birthdays that are both YYYYMMDD and YYYY-MM-DD, and Connect You (in my opinion), should transform both to YYYY-MM-DD if applicable.

### Other details

Crashed with YYYYMMDD, but fixed by simply resetting the birthday

https://paste.crdroid.net/r63Yef

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/you-apps/ConnectYou/issues/430#issue-2712102854

@apprehensions, yeah, ISO 8601 should be the internal normalised format. There's not much point to storing whatever the user enters and then trying to parse it, since no other format makes much sense (not even RFC 3339 / 9557).

**Comment by Bnyro** [comment]
Connect You won't automatically overwrite existing data (except if you re-save the contact), but it's now able to also handle malformed dates.
